kotlin-result/src/commonTest/kotlin/com/github/michaelbull/result/MapTest.kt

```diff
@@ -192,7 +192,7 @@ class MapTest {
 
         @Test
         fun returnsTransformedErrorIfNull() {
-            val result = Ok(null).toErrorIfNull { "a" }
+            val result: Result<Nothing, String> = Ok(null).toErrorIfNull { "a" }
```

Comment:
This would not have compiled before because the return type of `toErrorIfNull` was `Result<Nothing?, String>` which should not be possible.